### PR TITLE
MW-816/add user mob x

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 1]
```diff
@@ -280,11 +280,11 @@ export const WayPage = observer((props: WayPageProps) => {
             wayTags: way.wayTags,
           });
 
-          const updatedWayUuids = isWayExistInCollection
+          const updatedWays = isWayExistInCollection
             ? userCollection.ways.filter(wayPreview => wayPreview.uuid !== way.uuid)
             : userCollection.ways.concat(updatedWay);
 
-          return {...userCollection, wayUuids: updatedWayUuids};
+          return new WayCollection({...userCollection, ways: updatedWays});
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:287]: !

Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/WayPage.tsx]: ?

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 2]
```diff
@@ -449,7 +463,14 @@ export const UserPage = observer((props: UserPageProps) => {
                   uuid: userPageOwner.uuid,
                   description,
                 },
-                setUser: setUserPreviewPartial,
+
+                /**
+                 * Update user
+                 */
+                setUser: (uuser: PartialWithUuid<User>) => {
+                  setUser({...userPageOwner, ...uuser});
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx]: Done

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 3]
```diff
@@ -378,7 +385,14 @@ export const UserPage = observer((props: UserPageProps) => {
                   uuid: userPageOwner.uuid,
                   email,
                 },
-                setUser: setUserPreviewPartial,
+
+                /**
+                 * Update user
+                 */
+                setUser: (uuser: PartialWithUuid<User>) => {
+                  setUser({...userPageOwner, ...uuser});
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx]: Done

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 4]
```diff
@@ -298,7 +298,14 @@ export const UserPage = observer((props: UserPageProps) => {
                     uuid: userPageOwner.uuid,
                     name,
                   },
-                  setUser: setUserPreviewPartial,
+
+                  /**
+                   * Update user
+                   */
+                  setUser: (uuser: PartialWithUuid<User>) => {
+                    setUser({...userPageOwner, ...uuser});
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx]: Done

File: mw-webapp/src/component/header/Header.cy.tsx [hunk 5]
```diff
@@ -16,6 +19,10 @@ const HeaderTest = () => {
       <Header
         dataCy={HEADER_CY}
         user={testUserPreview}
+        setLanguage={() => {}}
```
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/component/header/Header.cy.tsx]: Added

**Pull request merged**